pkg/workflows/wasm/runner_wasip1.go

```diff
@@ -46,6 +50,22 @@ func NewRunner() *Runner {
 
 			os.Exit(code)
 		},
+		SDK: Runtime{
+			Logger: logger.NewWithSync(&wasmWriteSyncer{
+				logFn: log,
+			}),
+		},
 		args: os.Args,
 	}
 }
+
+type wasmWriteSyncer struct {
+	logFn func(respptr unsafe.Pointer, respptrlen int32)
+}
+
+// Write is used to proxy log requests from the WASM binary back to the host
+func (wws *wasmWriteSyncer) Write(p []byte) (n int, err error) {
+	ptr, ptrlen := bufferToPointerLen(p)
+	wws.logFn(ptr, ptrlen)
```

Comment:
nit/ do we actually swap this out? Or could it call `log` directly?
```suggestion
	log(ptr, ptrlen)
```

Reply:
we are not changing it, Im updating it. thanks!